packages/vue-renderer/src/renderers/ssr.js

```diff
@@ -180,7 +180,11 @@ export default class SSRRenderer extends BaseRenderer {
     if (renderContext.staticAssetsBase) {
       const preloadScripts = []
       renderContext.staticAssets = []
-      const { staticAssetsBase, url, nuxt, staticAssets } = renderContext
+      const { staticAssetsBase, nuxt, staticAssets } = renderContext
+      let url = new $URL(renderContext.url).pathname
+      if (!this.options.router.trailingSlash) {
+        url = withoutTrailingSlash(url) || '/'
```

Comment:
Note for later: we might handle empty string situation with ufo later since empty string is not valid path...